src/chainstate/burn/db/burndb.rs

```diff
@@ -421,363 +439,150 @@ const BURNDB_SETUP : &'static [&'static str]= &[
-        let args: &[&dyn ToSql] = &[tip_burn_header_hash, burn_header_hash, stacks_block_hash, &u64_to_sql(stacks_block_height)?];
-        tx.execute("INSERT OR REPLACE INTO canonical_accepted_stacks_blocks (tip_burn_block_hash, burn_block_hash, stacks_block_hash, block_height) VALUES (?1, ?2, ?3, ?4)", args)
-            .map_err(db_error::SqliteError)?;
-        Ok(())
+    pub fn conn_view<'b>(&'b self, view_sortition: &SortitionId) -> SortitionHandleConn<'b> {
+        let mut conn_view = self.as_conn();
+        conn_view.context.chain_tip = view_sortition.clone();
+        conn_view
     }
+}
 
-    /// Get an accepted stacks block header in a fork whose chain tip has not yet committed
-    /// to it.
-    fn get_accepted_stacks_block_pointer(conn: &Connection, tip_burn_header_hash: &BurnchainHeaderHash, stacks_block_hash: &BlockHeaderHash) -> Result<Option<AcceptedStacksBlockHeader>, db_error> {
-        let args : &[&dyn ToSql] = &[tip_burn_header_hash, stacks_block_hash];
-        let mut rows = query_rows(conn, "SELECT * FROM canonical_accepted_stacks_blocks WHERE tip_burn_block_hash = ?1 AND stacks_block_hash = ?2", args)?;
-        let len = rows.len();
-        match len {
-            0 => Ok(None),
-            1 => Ok(rows.pop()),
-            _ => {
-                panic!("BUG: the same Stacks block {} shows up twice or more in the same burn chain fork (whose tip is {})", stacks_block_hash, tip_burn_header_hash);
-            }
-        }
+impl <'a> SortitionDBTx <'a> { 
+    // PoX todo: this acceptance tracking will need to change once the StacksChainView is integrated
+    pub fn set_stacks_block_accepted_stubbed(&mut self, burn_header_hash: &BurnchainHeaderHash, parent_stacks_block_hash: &BlockHeaderHash,
+                                             stacks_block_hash: &BlockHeaderHash, stacks_block_height: u64) -> Result<(), db_error> {
+        let chain_tip = SortitionDB::get_canonical_burn_chain_tip_stubbed(self)?;
+        self.set_stacks_block_accepted_at_tip(&chain_tip, burn_header_hash, parent_stacks_block_hash, stacks_block_hash, stacks_block_height)
     }
 
     /// Mark an existing snapshot's stacks block as accepted at a particular burn chain tip, and calculate and store its arrival index.
     /// If this Stacks block extends the canonical stacks chain tip, then also update the memoized canonical
     /// stacks chain tip metadata on the burn chain tip.
-    fn set_stacks_block_accepted_at_tip<'a>(tx: &mut BurnDBTx<'a>, burn_tip: &BlockSnapshot, burn_header_hash: &BurnchainHeaderHash, parent_stacks_block_hash: &BlockHeaderHash, stacks_block_hash: &BlockHeaderHash, stacks_block_height: u64) -> Result<(), db_error> {
-        let arrival_index = BurnDB::get_arrival_index(tx)?;
+    fn set_stacks_block_accepted_at_tip(&mut self, burn_tip: &BlockSnapshot, burn_header_hash: &BurnchainHeaderHash,
```

Comment:
The DB operations in this method should occur within a transaction.  That doesn't appear to be happening here.  If updates to the DB aren't atomic, then it could cause the relayer and RPC threads to report bad or invalid information.

Reply:
`self` here is a `SortitionDBTx` -- which can only be constructed by making a rusqlite::Transaction,

Reply:
Gotcha :+1: